File: leetcode/subSets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function (nums) {
	let rs = [];
	let temp = [];
	const backtrack = (start) => {
		rs.push([...temp]);
		for (let i = start; i < nums.length; i++) {
			temp.push(nums[i]);
			backtrack(i + 1);
			temp.pop();
		}
	};
	backtrack(0);
	return rs;
};

subsets([1,2,3])

File: leetcode/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        return subsets(nums)


def subsets(nums):
    n = len(nums)
    rs = []
    temp = []

    def backtrack(start):
        rs.append(temp[:])
        for i in range(start, n):
            if i > start and nums[i] == nums[i - 1]:
                continue
            temp.append(nums[i])
            backtrack(i + 1)
            temp.pop()
    backtrack(0)
    return rs
